### DISCUSSION TITLE
v0.91.1-beta

### DISCUSSION THREAD

**Discussion by eliandoran** [TriliumNext/Trilium#6020] [Releases]
💡 Key highlights
-----------------

*   The launcher bar can now be placed at the top instead of the left of the screen, with a full-width tab bar. See [#654](https://github.com/TriliumNext/Notes/pull/654) for more information.
*   [A new theme is introduced called “TriliumNext"](https://github.com/TriliumNext/Notes/pull/661) by @adoriandoran and @eliandoran. Settings on this theme are now displayed in a card layout. Redesigned tree action buttons.
*   [Native title bar buttons](https://github.com/TriliumNext/Notes/pull/702) when native title bar is off for Windows and macOS.
    
*   On Windows, on the TriliumNext theme there are window transparency effects if running under Windows, see [Mica (Windows transparency effects)](https://github.com/TriliumNext/Notes/pull/717) for screenshots. Unfortunately, it is plagued by a few bugs with Electron (the library we are using for the desktop application), see the link for more information.
    
*   Updated Docker builds to Node 22. **Be on the lookout for any potential issues, especially for less-tested platforms such as ARM.**
    

🐞 Bugfixes
-----------

*   [Syntax highlight in code blocks only in edit mode](https://github.com/TriliumNext/Notes/issues/668)
*   To do lists not preserved on safe import.
*   [File type video with emoji in name will break the preview](https://github.com/TriliumNext/Notes/issues/757)
    
*   [Attribute "jump to" for internal links not working anymore](https://github.com/TriliumNext/Notes/issues/732)
    

✨ Improvements
--------------

*   [Add label definition type for standalone time](https://github.com/TriliumNext/Notes/pull/653) by @mm21
    
*   [Remove resizer from backend log text area](https://github.com/TriliumNext/Notes/pull/670) by @meichthys
*   Use CodeMirror for backend log
*   Support for ELK layout in Mermaid diagrams (sponsored by @perfectra1n). See the relevant “ELK layout engine” in [Mermaid diagrams](https://triliumnext.pikapod.net/share/mermaid-diagrams) for more information.
*   [Make exact matches rise higher up in search](https://github.com/TriliumNext/Notes/pull/678) by @perfectra1n
*   [Change layout of promoted attributes](https://github.com/TriliumNext/Notes/pull/538)
*   [Add full text search in autocomplete](https://github.com/TriliumNext/Notes/pull/659) by @SiriusXT
*   [Importing single HTML file: prefer html title over filename](https://github.com/TriliumNext/Notes/pull/598) by @maphew
*   [Support page breaks for printing](https://github.com/TriliumNext/Notes/pull/692)
*   [user configurable list of allowed html tags in import](https://github.com/TriliumNext/Notes/pull/601) by @maphew
*   General HTML support; now the editor supports more HTML tags and displays them appropriately.
    *   Currently the import function is the only way to benefit from them, or modifying the source code manually by changing the note type.
*   Change default icons for “Go to Previous Note” and “Go to next note” launcher buttons.
*   Move toggle sidebar button near tab bar on horizontal layout.
*   Mind map notes now are full-width by default (same as Canvas, for example).
*   Tabs can now display the icon notes instead of only workspace icons (this is the default behaviour of the TrilumNext theme, can be added to any other theme via `--tab-note-icons: true;` to `:root`).
*   [New Features for note map](https://github.com/TriliumNext/Notes/pull/700) by @CobriMediaJulien
*   [Add supported link protocols](https://github.com/TriliumNext/Notes/pull/694) by @SiriusXT
*   [add Mind Elixir plugin @mind-elixir/node-menu](https://github.com/TriliumNext/Notes/issues/723)
    
*   [Format dates and times](https://github.com/TriliumNext/Notes/pull/741) by @adoriandoran
    
*   [Expose cheerio in backend script api](https://github.com/TriliumNext/Notes/pull/758) by @kleutzinger 
*   [Format note paths](https://github.com/TriliumNext/Notes/pull/759) by @adoriandoran
    
*   [add remark about UFW issues](https://github.com/TriliumNext/Notes/pull/767) by @pano9000
*   [Code Block language support for Terraform (HCL)](https://github.com/TriliumNext/Notes/issues/739)
    
*   [Display the fixed toolbar on multiple lines](https://github.com/TriliumNext/Notes/issues/729) (adjustable in settings)
    
*   [Hide `internalLink` from `Jump To Note` dialog](https://github.com/TriliumNext/Notes/issues/713)
    

🌍 Internationalization
-----------------------

*   [Chinese translation improvements (& some new messages translated)](https://github.com/TriliumNext/Notes/pull/673) by @Nriver
*   [Add translation for doc notes](https://github.com/TriliumNext/Notes/pull/677) by @Nriver
*   [Spanish improvements](https://github.com/TriliumNext/Notes/pull/695/files) by @hasecilu
*   [Traditional Chinese support](https://github.com/TriliumNext/Notes/pull/698) by @dwong33
*   Translate import toast messages.
*   Translate hidden notes (including options).
*   Romanian improvements.
*   [Brazillian Portuguese](https://github.com/TriliumNext/Notes/pull/740) by @Nertonm work in progress (only server translations for now)
    
*   Translate note autocomplete.
    

⬆️ Library updates
------------------

*   i18next, i18next-http-backend, i18next-fs-backend
*   force-graph v1.47.0
*   express 4.2.12
*   mermaid 11.4.1
*   axios 1.7.9
*   katex 0.16.14
*   https-proxy-agent 7.0.6
*   better-sqlite3 11.6.0
*   electron 31.3.1 -> 33.2.1
*   jimp 0.22.12 -> 1.6.0
*   mind-elixir 4.3.3
*   Development tools
    *   electron-forge
    *   Docker Node.js updated to v20.18.1
    *   GitHub Actions workflow actions
    *   webpack v5.97.1
    *   TypeScript + types
    *   yargs
    *   helmet
    *   marked 15.0.3
    *   is-svg 5.1.0
    *   image-type 5.2.0
    *   eslint v9.16.0
    *   jsdom v25.0.1
    *   compression v1.7.5

<hr /><em>This discussion was created from the release <a href='https://github.com/TriliumNext/Notes/releases/tag/v0.91.1-beta'>v0.91.1-beta</a>.</em>

**Comment by Yohan-Janolin** [comment]
As a Trilium user for many, many years, I just wanted to thank the entire TriliumNext team for your fantastic work.
Your releases are truly top-notch, and I'm absolutely thrilled to see each new one come out.
Keep up the good work!

**Comment by ups7** [comment]
Fixed formatting bar, modern UI/UX/theme, and better mobile usability (and a native app, if at all possible/feasible) were the things I thought would make Trilium even greater. It's amazing to see that you've already ticked the first one off, and the second one is coming!

Thank you to all the developers for taking the already great Trilium to the Next level!

**Comment by hasecilu** [comment]
Is there a way to force the dark theme always? I only access to it via Firefox and because I don't have any type of light/dark theme variable set (Openbox, Hyprland)  the theme is always light. It plays very nice with Dark Reader extension though.

**Reply by adoriandoran** [reply]
For the moment, there is no way to force the dark theme. But will be very soon.

**Reply by adoriandoran** [reply]
It's on the way in #800.

**Reply by hasecilu** [reply]
That's great news, thanks!

**Comment by perfectra1n** [comment]
Please let me know if anyone has any suggestions to search improvements, or any examples / queries that don't return what they would expect.

**Reply by maphew** [reply]
I've been thinking about the worthiness of rank ordering or weighting certain trees. 

For example after importing static html with assets saved in a sub-dir, the media are in a note named _"Page-name files"_. Most of the time but not all the time when using search or jump to I want "Page-name", but don't want "Page-name files" near the top of results.

We could add a _searchWeight_ attribute but that would get messy and complicated quickly I think.
A simpler _searchExclude_ maybe. But then there'd need to be a "no I really mean everything" search.

We could sort search results by the relative position of the branch in the root: top branches are returned 1st, bottom branches last. It has the advantage of being intuitive and easy to communicate. It would exert a pressure on the relatively free-form organization that is one Trilium's big attractions, so I wonder if the idea is wise.

The UX of Jump-To is sometimes annoying in that after typing a search phrase if you arrow cursor through the results list it replaces the search text. So if you want to try refine you need to delete and start over. However it's really handy that arrowing does this when the search results are what you want! Maybe add undo? So `Ctrl-Z`, would return to the original search phrase? (And even cycle through all the phrases if a few things have been tried.)

These are just thoughts at this point. Maybe there are better ways to address the source annoyances.

**Reply by CobriMediaJulien** [reply]
@perfectra1n i´ve seen the script to  weight the results. What do you think about adding the feature to set your own weighting based on your search results?



I would suggest we open a discussion and collect ideas and discuss  improvements and possible implementation ideas.

**Reply by CobriMediaJulien** [reply]
@perfectra1n seems like highlighting words doesn´t work if you specify your search with e.g. 
note.content = 'somestring'

would you like to take this one? I can open an improvement request (not sure if it´s a bug).

**Reply by perfectra1n** [reply]
Do you have any pictures or examples of what you're referring to?

**Reply by perfectra1n** [reply]
If this is what you're referring to - having the text highlight from an "advanced" search, yeah that'll be an improvement request as it hasn't been done before. This image is from Zadam's Trilium, which I believe showcases what you're asking about:
![image](https://github.com/user-attachments/assets/8e5f16d9-c357-460b-9e4f-3b222d47fb74)

**Reply by CobriMediaJulien** [reply]
Yep. I mean it´s implemented for the normal search but not advanced quieres:

![image](https://github.com/user-attachments/assets/7fa11719-ef36-47bf-8523-3480c9ef3ff0)


Would be awesome to have it for advanced queries because u can also search with regex. But maybe thats too much work. Haven´t looked into the code yet.

**Comment by diyoyo** [comment]
Thanks a lot for this release.
The new theme is quite satisfying 😃 👍 

A few comments so far:
- Thanks for fixing #732 
- Made a comment in #538 : The "one column" thing is not suitable for very long lists of attributes.
- New bug (should I open a ticket since it's beta?) : the promoted attributes are no longer collapsible

**Reply by eliandoran** [reply]
@diyoyo ,

Thanks for the feedback!

> New bug (should I open a ticket since it's beta?) : the promoted attributes are no longer collapsible

Any suggestions on how to implement it? The goal of the promoted attributes change was to get rid of it from the promoted section, any UX/UI ideas would be appreciated.

**Reply by mm21** [reply]
IMHO this change of moving promoted attributes out of the ribbon is the way to go. Personally I prefer having the whole ribbon at the bottom (like in https://github.com/eliandoran/breeze-trilium) -- maybe the exact placement could just be left up to the theme.

**Comment by maphew** [comment]
The beta download is almost 100mb larger than stable. Is that expected?

**Reply by perfectra1n** [reply]
If you want fancy fonts, you've gotta pay the price 😂
![image](https://github.com/user-attachments/assets/e117eb2d-de14-4cfc-8a5c-8d75b00629fb)

**Reply by eliandoran** [reply]
> If you want fancy fonts, you've gotta pay the price 😂
> 
> ![image](https://github.com/user-attachments/assets/e117eb2d-de14-4cfc-8a5c-8d75b00629fb)
> 
> 

Oops.
